jwala-services/src/main/java/com/cerner/jwala/control/jvm/command/JvmCommandFactory.java

```diff
@@ -46,56 +44,105 @@
+        });
+        commands.put(JvmControlOperation.THREAD_DUMP.getExternalValue(), jvm -> {
+            final String threadDumpScriptName = THREAD_DUMP_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(jvm, threadDumpScriptName);
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm), getExecCommandForThreadDump(threadDumpScriptName, jvm)));
+        });
+        commands.put(JvmControlOperation.HEAP_DUMP.getExternalValue(), jvm -> {
+            final String heapDumpScriptName = HEAP_DUMP_SCRIPT_NAME.getValue();
+            checkExistsAndCopy(jvm, heapDumpScriptName);
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm), getExecCommandForHeapDump(heapDumpScriptName, jvm)));
+        });
+        commands.put(JvmControlOperation.DEPLOY_CONFIG_ARCHIVE.getExternalValue(), jvm -> {
+            checkExistsAndCopy(jvm, DEPLOY_CONFIG_ARCHIVE_SCRIPT_NAME.getValue());
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm), getExecCommandForDeploy(jvm)));
+        });
+        commands.put(JvmControlOperation.INSTALL_SERVICE.getExternalValue(), jvm -> {
+            checkExistsAndCopy(jvm, INSTALL_SERVICE_SCRIPT_NAME.getValue());
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm), getExecCommandForInstallService(jvm)));
+        });
+        commands.put(JvmControlOperation.DELETE_SERVICE.getExternalValue(), jvm -> {
+            checkExistsAndCopy(jvm, DELETE_SERVICE_SCRIPT_NAME.getValue());
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm), getExecCommandForDeleteService(jvm)));
+        });
+        commands.put(JvmControlOperation.CHECK_SERVICE_STATUS.getExternalValue(), jvm -> {
+            checkExistsAndCopy(jvm, SERVICE_STATUS_SCRIPT_NAME.getValue());
+            return remoteCommandExecutorService.executeCommand(new RemoteExecCommand(getConnection(jvm),getExecCommandForCheckServiceStatus(jvm)));
+        });
 
     }
 
+    private void checkExistsAndCopy(Jvm jvm, String scriptName) {
+        final String destAbsolutePath = getFullPathScript(jvm, scriptName);
+        CommandOutput fileExistsResult = binaryDistributionControlService.checkFileExists(jvm.getHostName(), destAbsolutePath);
+        if (!fileExistsResult.getReturnCode().wasSuccessful()) {
+            copyScriptToRemoteDestination(jvm, scriptName, destAbsolutePath);
+        } else {
+            LOGGER.info("{} already exists. Continue with script execution", scriptName);
+        }
+    }
+
+    private void copyScriptToRemoteDestination(Jvm jvm, String scriptName, String destAbsolutePath) {
+        LOGGER.info("{} does not exist at remote location. Performing secure copy.", scriptName);
+
+        final String destDir = new File(getFullPathScript(jvm, scriptName)).getParent();
```

Comment:
You are right, this will work on any OS, the issue will be it generates the OS specific paths. So jwala on Linux(or Mac) will not be able to Windows or vice versa